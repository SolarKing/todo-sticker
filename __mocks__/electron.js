// @flow
let listeners: Map<string, Function> = new Map();

class IpcMain {   

    on(channel: string, listener: Function) {
        listeners.set(channel, listener);
    }
    
}

class IpcRenderer {   

    send(channel: string, ...args: Array<mixed>) {
        const listener = listeners.get(channel);
        if (listener != null) {
            listener({}, ...args);
        }
    }
    
}

class BrowserWindow {
    _listeners: Map<string, Function> = new Map();

    webContents = {
        openDevTools() {

        },
        on(channel: string, listener: Function) {
            listeners.set(channel, listener);
        },
        emit(channel: string, ...args: Array<mixed>) {
            const listener = listeners.get(channel);
            if (listener != null) {
                listener(...args);
            }
        },
        destroy() {

        }
    }

    constructor() {
        this._listeners = new Map();
    }
    isMaximized() {
        return false;
    }
    loadURL() { }
    close() { }
    maximize() { }
    unmaximize() { }
    minimize() { }

    on(channel: string, listener: Function) {
        this._listeners.set(channel, listener);
    }

    emit(channel: string, ...args: Array<mixed>) {
        const listener = this._listeners.get(channel);
        if (listener != null) {
            listener(...args);
        }
    }

    static getFocusedWindow(): BrowserWindow {
        return new BrowserWindow();
    }

}

const ipcMain = new IpcMain();
const ipcRenderer = new IpcRenderer();

export {
    ipcMain,
    ipcRenderer,
    BrowserWindow
};