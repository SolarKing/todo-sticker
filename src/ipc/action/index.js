// @flow

export class TodoDescriptionChangedIPC {
    id: string;
    description: string;
}