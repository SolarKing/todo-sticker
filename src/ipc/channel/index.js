// @flow
export const IpcChannels = {
    noteDescriptionSend: 'noteDescriptionSend',
};