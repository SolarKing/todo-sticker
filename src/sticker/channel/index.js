// @flow
export const IpcChannels = {
    noteDescriptionSend: 'noteDescriptionSend',
    todoDescriptionChanged: 'todoDescriptionChanged',
    closeTodoNote: 'closeTodoNote'
};