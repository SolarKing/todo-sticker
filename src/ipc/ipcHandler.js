// @flow
import { ipcMain, BrowserWindow, WebContents, Electron } from 'electron';
import AddNote from './action/addNote';

declare var __dirname: any;

export default class IpcHandler {

    notes: Map<string, WebContents>;
    _createNoteAndSendDescription: Function;

    constructor() {
        this.notes = new Map();
        this._createNoteAndSendDescription = this._createNoteAndSendDescription.bind(this);
    }

    registerListener() {
        ipcMain.on(AddNote.ipcChannel, this._createNoteAndSendDescription);
    }

    _createNoteAndSendDescription(event: Electron.event, addNote: AddNote) {
        let win: BrowserWindow = new BrowserWindow({width: 400, height: 300});
        win.loadURL(`file://${__dirname}/../../dist/note/index.html`);
        win.webContents.openDevTools();

        win.webContents.on('dom-ready', () => {
            win.webContents.send('noteDescription', addNote.noteDescription);
        });
        this.notes.set(addNote.id, win.webContents);
    }
}
