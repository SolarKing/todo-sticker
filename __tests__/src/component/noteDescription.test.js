// @flow
import React from 'react';
import { shallow, type ShallowWrapper } from 'enzyme';
import NoteDescription from '../../../src/component/note/noteDescription';
import { IpcChannels } from '../../../src/ipc/channel';
import { ipcRenderer } from 'electron';

jest.mock('electron', () => {
    let _listener;
    let ipcRenderer = {
        on: (channel: string, listener: Function) => {
            _listener = listener;
        },
        send: (channel: string, message: string) => {
            _listener({}, message);
        }
    };
    return {
        ipcRenderer,
    };
});

describe('NodeDescription', () => {
    let wrapper: ShallowWrapper;

    beforeAll(() => {
        wrapper = shallow(
            <NoteDescription/>
        );
    });

    it('should render correct', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should set description when ipcRenderer send new message', () => {
        ipcRenderer.send(IpcChannels.noteDescriptionSend, '123');
        expect(wrapper.state().description).toEqual('123');
    });

});
