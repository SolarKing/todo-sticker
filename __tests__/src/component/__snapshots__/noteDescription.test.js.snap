// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeDescription should render correct 1`] = `<noteDescription__NoteDescriptionDiv />`;
